### PR TITLE
Image upload for Items

### PR DESCRIPTION
**Pull request by claussmann**
You have to upload an image when creating an item

### CODE REVIEW

File: src/main/java/de/hhu/propra/sharingplatform/dao/ImageService.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+@Component
+public class ImageService {
+
+    @Autowired
+    private HttpServletRequest request;
+
+    public void store(MultipartFile file, String filename) {
+        if (file.getContentType().equals("image/png") || file.getContentType().equals("image/jpeg")) {
+            try {
+                File out = ResourceUtils.getFile("src/main/resources/static/images/" + filename);
```
Comment (claussmann) [inline src/main/java/de/hhu/propra/sharingplatform/dao/ImageService.java]: This one is a little hacky... Better Idears are welcome!!! What happens when we put it in a jar? location will change...

File: src/main/java/de/hhu/propra/sharingplatform/dao/ImageService.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import java.net.URL;
+
+@Component
+public class ImageService {
+
+    @Autowired
+    private HttpServletRequest request;
+
+    public void store(MultipartFile file, String filename) {
+        if (file.getContentType().equals("image/png") || file.getContentType().equals("image/jpeg")) {
```
Comment (claussmann) [inline src/main/java/de/hhu/propra/sharingplatform/dao/ImageService.java]: Only jpg and png allowed.

File: src/main/java/de/hhu/propra/sharingplatform/dao/ImageService.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            try {
+                File out = ResourceUtils.getFile("src/main/resources/static/images/" + filename);
+                out.createNewFile();
+                FileOutputStream fos = new FileOutputStream(out);
+                fos.write(file.getBytes());
+                fos.close();
+            } catch (IOException exception) {
+                throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR,
```
Comment (claussmann) [inline src/main/java/de/hhu/propra/sharingplatform/dao/ImageService.java]: We can make a better response if we check the file size first. To big files should throw a different error

File: src/main/java/de/hhu/propra/sharingplatform/dao/ImageService.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
+package de.hhu.propra.sharingplatform.dao;
```
Comment (DirtyDan007) [inline src/main/java/de/hhu/propra/sharingplatform/dao/ImageService.java]: ImageService should be in service package

File: src/main/java/de/hhu/propra/sharingplatform/dao/ImageService.java (hunk 1 above)
Comment (Grimaldus12) [reply src/main/java/de/hhu/propra/sharingplatform/dao/ImageService.java]: I guess there is a way to set the path in the manifest file.

File: src/main/java/de/hhu/propra/sharingplatform/dao/ImageService.java (hunk 3 above)
Comment (Grimaldus12) [reply src/main/java/de/hhu/propra/sharingplatform/dao/ImageService.java]: Good idea

**Review by Grimaldus12 — APPROVED** [review]
Approved.

**Pull request merged**